src/pattern.coffee

```diff
@@ -102,6 +102,11 @@ class Pattern
     else if name is "$base"
       baseGrammar.getInitialRule()
     else
+      hashIndex = name.indexOf("#")
+      if hashIndex isnt -1
+        grammar = @registry.grammarForScopeName(name[0..hashIndex-1])
+        if grammar
+          return grammar.getRepository()[name[hashIndex+1..]]
       @grammar.addIncludedGrammarScope(name)
```

Comment:
I think it might be good to still call this line with the root scope name (the text before the `#`). This allows the tokenizer to re-run when a dependent grammar is loaded since grammars are loaded asynchronously.